pkg/core/gitadmin.go

```diff
@@ -0,0 +1,39 @@
+/**
+ * This is the administration interface to the git store
+ * It is currently implemented using gitea API
+ */
+package fuseml
+
+import (
+	"github.com/pkg/errors"
+
+	codeset "github.com/fuseml/fuseml-core/gen/codeset"
+	giteaadmin "github.com/fuseml/fuseml-core/pkg/core/gitea"
+)
+
+type GitAdmin struct {
```

Comment:
This could be easily changed to fully apply the dependency inversion principle here. If you code this to be an interface instead of a struct, then you can instantiate a GiteaAdminClient object as a concrete implementation of GitAdmin and pass it to the codeset store to be used as a GitAdmin interface object. The way to tell if this principle works is when neither this file nor the codeset_store.go file has any direct dependencies to the GiteaAdminClient type.

Reply:
Here's a link to a post concerning a dependency inversion problem that is very similar to ours: https://medium.com/@2hamed/dependency-isolation-in-go-afab00b5e4c9